frame/nfts/src/lib.rs

```diff
@@ -630,8 +647,15 @@ pub mod pallet {
 
 			Self::do_transfer(collection, item, dest, |collection_details, details| {
 				if details.owner != origin && collection_details.admin != origin {
-					let approved = details.approved.take().map_or(false, |i| i == origin);
+					let approved = details.approvals.contains_key(&origin);
 					ensure!(approved, Error::<T, I>::NoPermission);
```

Comment:
Given the new code these lines are not needed?
```suggestion
```

Reply:
Good catch!